### ISSUE TITLE
build node tries to add non-existent docker image to pyrsia

### ISSUE THREAD

**Issue by sbtaylor15** [pyrsia/pyrsia#1193]
When build-docker.sh tries to curl the manifest for a non-existent image it returns `404 page not found` in the temp manifest file.  The subsequent jq command returns null.  Next the build node tries to add a `null` image to pyrsia which causes errors with the blockchain and artifact manager.

## image name is spelled wrong which causes the error

```
#######################################################
+ BUILD_ID=8d15ab4e-7f96-48d8-bd31-36d3541b7a95
+ IMAGE_NAME=library/apline
+ IMAGE_REFERENCE=3.16
+ mkdir -p artifacts
+ mkdir -p docker
+ cd docker
+ curl+ jq -r .token
 https://auth.docker.io/token?client_id=Pyrsia&service=registry.docker.io&scope=repository:library/apline:pull
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
^M  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0^M100  4055    0  4055    0     0  25343      0 --:--:-- --:--:-- --:--:-- 25343
+ DOCKER_IO_TOKEN=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsIng1YyI6WyJNSUlDK1RDQ0FwK2dBd0lCQWdJQkFEQUtCZ2dxaGtqT1BRUURBakJHTVVRd1FnWURWUVFERXp0U1RVbEdPbEZNUmpRNlEwZFFNenBSTWtWYU9sRklSRUk2VkVkRlZUcFZTRlZNT2taTVZqUTZSMGRXV2pwQk5WUkhPbFJMTkZNNlVVeElTVEFlRncweU1qQXhNVEF5TWpJeE5EbGFGdzB5TXpBeE1qVXlNakl4TkRsYU1FWXhSREJDQmdOVkJBTVRPMUJaUTFJNlNWQmFRanBJUWxGWE9qZE1SVms2UWtGV1FqcEhTRGRhT2xWSVZ6TTZVa3RMVWpwRE4wNHpPbGxOTkZJNlNWaE5TRHBLVkZCQ01JSUJJakFOQmdrcWhraUc5dzBCQVFFRkFBT0NBUThBTUlJQkNnS0NBUUVBbE5JOGFtMlBERnZzNndOeVl2d0dkZWZXVXJXQWdMZ3N2MzA2MnZycG5VNkN0WUpEZDZ2K2NEVG1FN1FlTllEaFMyaU1wU3djZkRCL2RFclEwdnhIZE4ycDIvODZmZy9TeWlIMnhmMGFVTjlDV1dud0JPaTIvS3hLditpbFNlQ01HYXRwRlg3SmYxcWI4N0Q5NUxOVDBvOU9OTmYxT3RidjY5ck9tL1RIVFh3clUvV3dTZlUyWktUbEw4SVRXRkRXN09ZK3hXdUJ0WUpteVhqcVpsaWRBbUNTdTdHY0Y0MVB5em9KTFFTMnJCdXJwOXc0cWgxMFk1bUNIcWdsaEI1Rk9aOUs0T2pUaVhUUHJFUk5WcnArUFVIR3JBYVRPRTBwQzgyUHBuWVZhNzNDUkdsMEdDdC9RckJwVjRpdmswdzF0eEtkV1NiSDNnRmtqZ2g1N0tOcDhRSURBUUFCbzRHeU1JR3ZNQTRHQTFVZER3RUIvd1FFQXdJSGdEQVBCZ05WSFNVRUNEQUdCZ1JWSFNVQU1FUUdBMVVkRGdROUJEdFFXVU5TT2tsUVdrSTZTRUpSVnpvM1RFVlpPa0pCVmtJNlIwZzNXanBWU0Zjek9sSkxTMUk2UXpkT016cFpUVFJTT2tsWVRVZzZTbFJRUWpCR0JnTlZIU01FUHpBOWdEdFNUVWxHT2xGTVJqUTZRMGRRTXpwUk1rVmFPbEZJUkVJNlZFZEZWVHBWU0ZWTU9rWk1WalE2UjBkV1dqcEJOVlJIT2xSTE5GTTZVVXhJU1RBS0JnZ3Foa2pPUFFRREFnTklBREJGQWlFQTdIY1VyVm1namo1cE01MXhZVHd2eGE1VnRqd2hub0dRZjFxTU52UGVHeVlDSUFwYm4vWFkvS1F5WWFWRnRjMWtsb0lmZzd4L3hlbkZhbkp4L0F2cURGdFgiXX0.eyJhY2Nlc3MiOltdLCJhdWQiOiJyZWdpc3RyeS5kb2NrZXIuaW8iLCJleHAiOjE2NjU2NzQwNjksImlhdCI6MTY2NTY3Mzc2OSwiaXNzIjoiYXV0aC5kb2NrZXIuaW8iLCJqdGkiOiJkY2tyX2p0aV9sXzZXcGtfQVM4T2I4UnFpOU9Hd3E0Q1k3WU09IiwibmJmIjoxNjY1NjczNDY5LCJzdWIiOiIifQ.hkgPY8kYm5K3-H0thEQW8r7pPFEwja0urjmQavl8Ajheu4Vi5JwLA2-one5VbmdRP-rWK8_PHtEmzc10D4MUsVLL_uiiLXXApYvDC4Hg4mCNF398fQpQoom_dCCZUslVt6oQ58kHfV3m0VAuyZWJkCvLevNCgJXJGLScIrQB6j51e0TTWLqIIcJkoWfbxlLNMpv0ncKYGtQ4xn7F4bQCGqXpUdSipkbGL_spYaglVyr9HZnbYoLtSqLC4nJpwXccN0mdEPQR8zw2NmvHo7xuolWGmKr4ltBGIe5GDmuCzWyCfus4j2GJ9vLnh6u2xYpG0iH6vQqYuCiD-PhRbyya_Q
+ curl -L https://registry-1.docker.io/v2/library/apline/manifests/3.16 -H Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsIng1YyI6WyJNSUlDK1RDQ0FwK2dBd0lCQWdJQkFEQUtCZ2dxaGtqT1BRUURBakJHTVVRd1FnWURWUVFERXp0U1RVbEdPbEZNUmpRNlEwZFFNenBSTWtWYU9sRklSRUk2VkVkRlZUcFZTRlZNT2taTVZqUTZSMGRXV2pwQk5WUkhPbFJMTkZNNlVVeElTVEFlRncweU1qQXhNVEF5TWpJeE5EbGFGdzB5TXpBeE1qVXlNakl4TkRsYU1FWXhSREJDQmdOVkJBTVRPMUJaUTFJNlNWQmFRanBJUWxGWE9qZE1SVms2UWtGV1FqcEhTRGRhT2xWSVZ6TTZVa3RMVWpwRE4wNHpPbGxOTkZJNlNWaE5TRHBLVkZCQ01JSUJJakFOQmdrcWhraUc5dzBCQVFFRkFBT0NBUThBTUlJQkNnS0NBUUVBbE5JOGFtMlBERnZzNndOeVl2d0dkZWZXVXJXQWdMZ3N2MzA2MnZycG5VNkN0WUpEZDZ2K2NEVG1FN1FlTllEaFMyaU1wU3djZkRCL2RFclEwdnhIZE4ycDIvODZmZy9TeWlIMnhmMGFVTjlDV1dud0JPaTIvS3hLditpbFNlQ01HYXRwRlg3SmYxcWI4N0Q5NUxOVDBvOU9OTmYxT3RidjY5ck9tL1RIVFh3clUvV3dTZlUyWktUbEw4SVRXRkRXN09ZK3hXdUJ0WUpteVhqcVpsaWRBbUNTdTdHY0Y0MVB5em9KTFFTMnJCdXJwOXc0cWgxMFk1bUNIcWdsaEI1Rk9aOUs0T2pUaVhUUHJFUk5WcnArUFVIR3JBYVRPRTBwQzgyUHBuWVZhNzNDUkdsMEdDdC9RckJwVjRpdmswdzF0eEtkV1NiSDNnRmtqZ2g1N0tOcDhRSURBUUFCbzRHeU1JR3ZNQTRHQTFVZER3RUIvd1FFQXdJSGdEQVBCZ05WSFNVRUNEQUdCZ1JWSFNVQU1FUUdBMVVkRGdROUJEdFFXVU5TT2tsUVdrSTZTRUpSVnpvM1RFVlpPa0pCVmtJNlIwZzNXanBWU0Zjek9sSkxTMUk2UXpkT016cFpUVFJTT2tsWVRVZzZTbFJRUWpCR0JnTlZIU01FUHpBOWdEdFNUVWxHT2xGTVJqUTZRMGRRTXpwUk1rVmFPbEZJUkVJNlZFZEZWVHBWU0ZWTU9rWk1WalE2UjBkV1dqcEJOVlJIT2xSTE5GTTZVVXhJU1RBS0JnZ3Foa2pPUFFRREFnTklBREJGQWlFQTdIY1VyVm1namo1cE01MXhZVHd2eGE1VnRqd2hub0dRZjFxTU52UGVHeVlDSUFwYm4vWFkvS1F5WWFWRnRjMWtsb0lmZzd4L3hlbkZhbkp4L0F2cURGdFgiXX0.eyJhY2Nlc3MiOltdLCJhdWQiOiJyZWdpc3RyeS5kb2NrZXIuaW8iLCJleHAiOjE2NjU2NzQwNjksImlhdCI6MTY2NTY3Mzc2OSwiaXNzIjoiYXV0aC5kb2NrZXIuaW8iLCJqdGkiOiJkY2tyX2p0aV9sXzZXcGtfQVM4T2I4UnFpOU9Hd3E0Q1k3WU09IiwibmJmIjoxNjY1NjczNDY5LCJzdWIiOiIifQ.hkgPY8kYm5K3-H0thEQW8r7pPFEwja0urjmQavl8Ajheu4Vi5JwLA2-one5VbmdRP-rWK8_PHtEmzc10D4MUsVLL_uiiLXXApYvDC4Hg4mCNF398fQpQoom_dCCZUslVt6oQ58kHfV3m0VAuyZWJkCvLevNCgJXJGLScIrQB6j51e0TTWLqIIcJkoWfbxlLNMpv0ncKYGtQ4xn7F4bQCGqXpUdSipkbGL_spYaglVyr9HZnbYoLtSqLC4nJpwXccN0mdEPQR8zw2NmvHo7xuolWGmKr4ltBGIe5GDmuCzWyCfus4j2GJ9vLnh6u2xYpG0iH6vQqYuCiD-PhRbyya_Q -H Accept: application/vnd.docker.distribution.manifest.v2+json -o manifest
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
^M  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0^M100   157  100   157    0     0   1226      0 --:--:-- --:--:-- --:--:--  1226
+ cat manifest
+ jq -r .config.digest
+ CONFIG_DIGEST=null
+ curl -L https://registry-1.docker.io/v2/library/apline/blobs/null -H Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsIng1YyI6WyJNSUlDK1RDQ0FwK2dBd0lCQWdJQkFEQUtCZ2dxaGtqT1BRUURBakJHTVVRd1FnWURWUVFERXp0U1RVbEdPbEZNUmpRNlEwZFFNenBSTWtWYU9sRklSRUk2VkVkRlZUcFZTRlZNT2taTVZqUTZSMGRXV2pwQk5WUkhPbFJMTkZNNlVVeElTVEFlRncweU1qQXhNVEF5TWpJeE5EbGFGdzB5TXpBeE1qVXlNakl4TkRsYU1FWXhSREJDQmdOVkJBTVRPMUJaUTFJNlNWQmFRanBJUWxGWE9qZE1SVms2UWtGV1FqcEhTRGRhT2xWSVZ6TTZVa3RMVWpwRE4wNHpPbGxOTkZJNlNWaE5TRHBLVkZCQ01JSUJJakFOQmdrcWhraUc5dzBCQVFFRkFBT0NBUThBTUlJQkNnS0NBUUVBbE5JOGFtMlBERnZzNndOeVl2d0dkZWZXVXJXQWdMZ3N2MzA2MnZycG5VNkN0WUpEZDZ2K2NEVG1FN1FlTllEaFMyaU1wU3djZkRCL2RFclEwdnhIZE4ycDIvODZmZy9TeWlIMnhmMGFVTjlDV1dud0JPaTIvS3hLditpbFNlQ01HYXRwRlg3SmYxcWI4N0Q5NUxOVDBvOU9OTmYxT3RidjY5ck9tL1RIVFh3clUvV3dTZlUyWktUbEw4SVRXRkRXN09ZK3hXdUJ0WUpteVhqcVpsaWRBbUNTdTdHY0Y0MVB5em9KTFFTMnJCdXJwOXc0cWgxMFk1bUNIcWdsaEI1Rk9aOUs0T2pUaVhUUHJFUk5WcnArUFVIR3JBYVRPRTBwQzgyUHBuWVZhNzNDUkdsMEdDdC9RckJwVjRpdmswdzF0eEtkV1NiSDNnRmtqZ2g1N0tOcDhRSURBUUFCbzRHeU1JR3ZNQTRHQTFVZER3RUIvd1FFQXdJSGdEQVBCZ05WSFNVRUNEQUdCZ1JWSFNVQU1FUUdBMVVkRGdROUJEdFFXVU5TT2tsUVdrSTZTRUpSVnpvM1RFVlpPa0pCVmtJNlIwZzNXanBWU0Zjek9sSkxTMUk2UXpkT016cFpUVFJTT2tsWVRVZzZTbFJRUWpCR0JnTlZIU01FUHpBOWdEdFNUVWxHT2xGTVJqUTZRMGRRTXpwUk1rVmFPbEZJUkVJNlZFZEZWVHBWU0ZWTU9rWk1WalE2UjBkV1dqcEJOVlJIT2xSTE5GTTZVVXhJU1RBS0JnZ3Foa2pPUFFRREFnTklBREJGQWlFQTdIY1VyVm1namo1cE01MXhZVHd2eGE1VnRqd2hub0dRZjFxTU52UGVHeVlDSUFwYm4vWFkvS1F5WWFWRnRjMWtsb0lmZzd4L3hlbkZhbkp4L0F2cURGdFgiXX0.eyJhY2Nlc3MiOltdLCJhdWQiOiJyZWdpc3RyeS5kb2NrZXIuaW8iLCJleHAiOjE2NjU2NzQwNjksImlhdCI6MTY2NTY3Mzc2OSwiaXNzIjoiYXV0aC5kb2NrZXIuaW8iLCJqdGkiOiJkY2tyX2p0aV9sXzZXcGtfQVM4T2I4UnFpOU9Hd3E0Q1k3WU09IiwibmJmIjoxNjY1NjczNDY5LCJzdWIiOiIifQ.hkgPY8kYm5K3-H0thEQW8r7pPFEwja0urjmQavl8Ajheu4Vi5JwLA2-one5VbmdRP-rWK8_PHtEmzc10D4MUsVLL_uiiLXXApYvDC4Hg4mCNF398fQpQoom_dCCZUslVt6oQ58kHfV3m0VAuyZWJkCvLevNCgJXJGLScIrQB6j51e0TTWLqIIcJkoWfbxlLNMpv0ncKYGtQ4xn7F4bQCGqXpUdSipkbGL_spYaglVyr9HZnbYoLtSqLC4nJpwXccN0mdEPQR8zw2NmvHo7xuolWGmKr4ltBGIe5GDmuCzWyCfus4j2GJ9vLnh6u2xYpG0iH6vQqYuCiD-PhRbyya_Q -o null.blob
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
^M  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0^M100    19  100    19    0     0    147      0 --:--:-- --:--:-- --:--:--   147
+ cat manifest
+ read b
+ jq -r .layers[].digest
jq: error (at <stdin>:1): Cannot iterate over null (null)
+ mv manifest null.blob ../artifacts/
```

**Comment by betarelease** [comment]
This should probably become a rust job instead of staying in shell now. So that we can test all these scenarios. Maybe create a rust background job which runs this and will only be used on authority node/build node. (so should not become part of main.rs)

**Comment by tiainen** [comment]
@sbtaylor15 I've added a commit to the build pipeline prototype repository: https://github.com/tiainen/pyrsia_build_pipeline_prototype/commit/fdfa3999358cbc51b22c3b4979cbf30ec2fe3977. The process will now exit when the request for fetching the docker manifest returns with a HTTP response code that is >= 400.

**Comment by betarelease** [comment]
Deployment tasks still pending on this @sbtaylor15

**Comment by sbtaylor15** [comment]
Pulled the latest build_pipeline_prototype on nightly and working as expected.  Will create a new issue if I run into errors on prod.
